# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/x86/MacroAssembler-x86-inl.h
# Commit: 6089b4360edc
# Full Hash: 6089b4360edcfb3a4bda506e37be8ca75a9f15a6
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 6
# Description:
#   Bug 1727084 - Memory64 - Non-atomic-RMW baseline operations. r=yury
#   
#   Implement scalar, SIMD, and atomic loads and stores for the baseline
#   compiler for x64, x86, arm64, and arm32.  Atomic RMW operations and
#   wait/notify are in the next patch.
# ==============================================================================

diff -r b9d32ef0ad83 -r 6089b4360edc js/src/jit/x86/MacroAssembler-x86-inl.h
--- a/js/src/jit/x86/MacroAssembler-x86-inl.h	Mon Oct 18 07:31:55 2021 +0000
+++ b/js/src/jit/x86/MacroAssembler-x86-inl.h	Mon Oct 18 07:31:55 2021 +0000
@@ -962,6 +962,12 @@
   j(Assembler::Overflow, fail);
 }
 
+void MacroAssembler::branchAdd64(Condition cond, Imm32 imm, Register64 dest,
+                                 Label* label) {
+  add64(imm, dest);
+  j(cond, label);
+}
+
 void MacroAssembler::branchTest32(Condition cond, const AbsoluteAddress& lhs,
                                   Imm32 rhs, Label* label) {
   test32(Operand(lhs), rhs);